recipes-ros/catkin/files/0001-cross-compilation.patch

```diff
@@ -0,0 +1,142 @@
+   set(_msg "WARNING: package '@PROJECT_NAME@' is deprecated")
+@@ -102,7 +111,19 @@ if(NOT "@PROJECT_CMAKE_CONFIG_INCLUDE_DIRS@ " STREQUAL " ")
+     set(_report "Report the problem to the maintainer '@PROJECT_MAINTAINER@' and request to fix the problem.")
+   endif()
+   foreach(idir ${_include_dirs})
+-    if(IS_ABSOLUTE ${idir} AND IS_DIRECTORY ${idir})
++    if(IS_ABSOLUTE ${idir} AND CMAKE_CROSSCOMPILING)
++      get_filename_component(idir_path ${idir} DIRECTORY)
++      get_filename_component(idir_name ${idir} NAME)
++      set(include_dir_path "include_dir_path-NOTFOUND")
++      # use find_path to locate absolute path nested under potential root path
++      find_path(include_dir_path ${idir_name}
++        PATHS ${idir_path}
++        NO_DEFAULT_PATH ${_find_root_path_both})
++      if(NOT include_dir_path)
++        message(FATAL_ERROR "Project '@PROJECT_NAME@' specifies '${idir}' as an include dir, which is not found.  It does not exist in any of the root filesystems.  Ask the maintainer '@PROJECT_MAINTAINER@' to fix it.")
++      endif()
++      set(include "${include_dir_path}/${idir_name}")
++    elseif(IS_ABSOLUTE ${idir} AND IS_DIRECTORY ${idir})
+       set(include ${idir})
+     elseif("${idir} " STREQUAL "@CATKIN_GLOBAL_INCLUDE_DESTINATION@ ")
+       get_filename_component(include "${@PROJECT_NAME@_DIR}/../../../@CATKIN_GLOBAL_INCLUDE_DESTINATION@" ABSOLUTE)
+@@ -123,16 +144,29 @@ foreach(library ${libraries})
+     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
+   elseif(TARGET ${library})
+     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
++  elseif(IS_ABSOLUTE ${library} AND CMAKE_CROSSCOMPILING)
++    get_filename_component(lib_path ${library} DIRECTORY)
++    get_filename_component(lib_name ${library} NAME)
++    set(lib "lib-NOTFOUND")
++    # use find_library to locate library given by absolute path nested under potential root path
++    find_library(lib ${lib_name}
++      PATHS ${lib_path}
++      NO_DEFAULT_PATH ${_find_root_path_both})
++    if(NOT lib)
++      message(FATAL_ERROR "Project '${PROJECT_NAME}' tried to find library '${library}'.  The library is neither a target nor built/installed properly.  Did you compile project '@PROJECT_NAME@'?  Did you find_package() it before the subdirectory containing its code is included?")
++    endif()
++    _list_append_unique(@PROJECT_NAME@_LIBRARY_DIRS ${lib_path})
++    list(APPEND @PROJECT_NAME@_LIBRARIES ${lib})
+   elseif(IS_ABSOLUTE ${library})
+     list(APPEND @PROJECT_NAME@_LIBRARIES ${library})
+   else()
+     set(lib_path "")
+-    set(lib "${library}-NOTFOUND")
++    set(lib "lib-NOTFOUND")
+     # since the path where the library is found is returned we have to iterate over the paths manually
+     foreach(path @PKG_CONFIG_LIB_PATHS@)
+       find_library(lib ${library}
+         PATHS ${path}
+-        NO_DEFAULT_PATH NO_CMAKE_FIND_ROOT_PATH)
++        NO_DEFAULT_PATH)
+       if(lib)
+         set(lib_path ${path})
```

Comment:
I presume the problems with `industrial-robot-client` and `moveit-kinematics` are due to this line. My patch contains this part
```
       if(lib)
-        set(lib_path ${path})
+        get_filename_component(lib_path ${lib} DIRECTORY)
         break()
       endif()

```
because the `NO_DEFAULT_PATH` option in `find_library` doesn't actually mean that the functions looks for a lib in the provided `path` only. It still might search in some other paths.

This actually reminds me that the PR needs to be tested for the case when the build host has some ROS packages installed and those packages' libs don't leak to bitbake. Like what happens if to build a package linking to `librostime.so` if there is an empty file `/opt/ros/indigo/lib/librostime.so` on the host.